pulp_python/app/utils.py

```diff
@@ -315,7 +315,7 @@ def find_artifact():
 
     content_artifact = content.contentartifact_set.first()
     artifact = find_artifact()
-    origin = settings.CONTENT_ORIGIN.strip("/")
+    origin = settings.CONTENT_ORIGIN.strip("/") if settings.CONTENT_ORIGIN else ""
```

Comment:
If `settings.CONTENT_ORIGIN` was None, wouldn't this result in a URL that starts with `/`?

Reply:
It might be better to do something more along the lines of:

```python
components = [prefix, base_path, content.filename]
if settings.CONTENT_ORIGIN:
    components.insert(0, settings.CONTENT_ORIGIN.strip("/"))
...
```

Which is the same way domain is handled more or less

Reply:
Or skip the `.insert()` and instead just build the list component by component.

```python
components = []
if settings.CONTENT_ORIGIN:
    components.append(settings.CONTENT_ORIGIN.strip("/"))
components.append(prefix)
if domain:
    components.append(domain.name)
components.append(base_path)
components.append(content.filename)
```

Which is probably simplest

Reply:
Ok I updated it to try to match what I did in `pypi/views`, basically try the `CONTENT_ORIGIN` else assume same fqdn and use `PYPI_API_HOSTNAME`, else set to empty string and hope the client interacting with the pypi API can handle relative urls.